lib/typing/decimal/decimal.go

```diff
@@ -4,30 +4,15 @@ import (
 	"github.com/cockroachdb/apd/v3"
 )
 
+const PrecisionNotSpecified int32 = -1
+
 // Decimal is Artie's wrapper around [*apd.Decimal] which can store large numbers w/ no precision loss.
 type Decimal struct {
 	precision int32
 	value     *apd.Decimal
 }
 
-const (
-	DefaultScale          int32 = 5
-	PrecisionNotSpecified int32 = -1
-	// MaxPrecisionBeforeString - if the precision is greater than 38, we'll cast it as a string.
-	// This is because Snowflake and BigQuery both do not have NUMERIC data types that go beyond 38.
-	MaxPrecisionBeforeString int32 = 38
-)
-
 func NewDecimalWithPrecision(value *apd.Decimal, precision int32) *Decimal {
-	scale := -value.Exponent
-	if scale > precision && precision != PrecisionNotSpecified {
-		// Note: -1 precision means it's not specified.
-
-		// This is typically not possible, but Postgres has a design flaw that allows you to do things like: NUMERIC(5, 6) which actually equates to NUMERIC(7, 6)
-		// We are setting precision to be scale + 1 to account for the leading zero for decimal numbers.
-		precision = scale + 1
```

Comment:
How come we don't need this?

Reply:
Because we have this exact logic in `NewDetails()` and we don't read from `Decimal.Precision()` except for tests.

Reply:
Got it. We should refactor the precision function then

```
func (d *Decimal) Precision() int32 {
	return NewDetails(d.precision, d.Scale()).Precision()
}
```

Reply:
I'd just as soon as kill `Precision`, we don't need it.